bin/main

```diff
@@ -1,111 +1,130 @@
+class Game
+  attr_reader :board, :move, :tic_tac_toe
+  attr_accessor :player1, :player2, :players, :loc, :locations, :usedd
+
+  def initialize
+    @board = %w[1 2 3 4 5 6 7 8 9]
+    @player_char = %w[X O]
+    @game = true
+    @players = []
+    @temp = ''
+    @locations = []
+    @loc = ''
+    @usedd = [2]
+    @move = 0
+    @tic_tac_toe = TicTacToe.new
+    @arb_i = 0
+  end
+
+  def gaming
+    @gamin = []
+    @store_players = create_players
+    puts "#{@store_players[0]} is going to play with X\n"
+    puts "#{@store_players[1]} is going to play with O\n\n"
+    puts "Great ! #{@store_players[0]} and #{@store_players[1]} let start the battle !!\n\n\n"
+
+    disply_board
+    i = 0
+    while i < 9
+      break unless take_input(@gamin)
+
+      i += 1
     end
-    print " #{array[position]} |"
-    position += 1
+    p tic_tac_toe.winner
   end
-  puts "\n---+---+---+"
-  puts "\n\n\n"
-end
 
-def check_input(array, player_char, player, loc, usedd, move)
-  while loc.is_a?(String)
-    print "#{player}, choose a position from (1-9) to place in the board: "
-    loc = gets.chomp
-    if loc =~ /[1-9]/ && loc.to_i.between?(1, 9) && loc.to_i <= array.length
-      loc = loc.to_i
-      loc = gets.chomp.to_i while taken?(loc, usedd)
-      array[loc - 1] = player_char
-      disply_board(array)
-    elsif loc.to_i.zero?
-      puts 'Zero is not part of the range (1-9)'
-    elsif move == array.length
-      puts "IT's a TIE !! "
-      break
-    elsif loc.to_i.negative?
-      puts 'No negatvie numbers allowed.'
+  def disply_board
+    position = 0
+    while position < @board.length
+      if (position % 3).zero?
+        puts ''
+        print "---+---+---+\n"
+      end
+      print " #{@board[position]} |"
+      position += 1
     end
+    puts "\n---+---+---+"
+    puts "\n\n\n"
   end
-end
 
-def check_empty_pseudo(player)
-  while player.empty?
-    print 'Enter a valid pseudo: '
-    player = gets.chomp
+  def create_players
+    @i = 0
+    while @i < 2
+      print "Enter pseudo for Player #{@i + 1}: "
+      input = gets.chomp
+      test = Player.new(input, @player_char[@i])
+      @players << test.name
+      if @tic_tac_toe.pseudo_empty?(@players[@i])
+        while @tic_tac_toe.pseudo_empty?(@players[@i])
+          puts "Don't leave player names empty"
+          print "Enter pseudo for Player #{@i + 1}: "
+          @players[@i] = gets.chomp
+        end
+      end
+      @i += 1
+    end
+    @players
   end
-  player
-end
 
-system('clear') || system('cls')
+  def take_input(usedd)
+    @i = 0
+    @usedd = usedd
+    @flag = true
+    while @temp.is_a?(String) && @i < 2 && @flag
+      print "#{@players[@i]}, choose a position from (1-9) to place in the board: "
+      @temp = gets.chomp
+      check_input(@temp, @players[@i])
+      win = @tic_tac_toe.arbiter(@board, @temp, @player_char[@i], players[@i])
+      @i += 1
+      @flag = false if win.length.positive?
+    end
+    @flag
+  end
```

Comment:
- [ ] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/single_responsibility.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

If I am not mistaken, you are taking a user input and also checking for a winner with the arbiter, please divide those steps into 2 different functions please. (If I am mistaken, ignore this comment.